File: app/src/main/java/com/example/movieapi/data/MovieRepo.java
package com.example.movieapi.data;

import com.example.movieapi.api.Description;
import com.example.movieapi.api.MovieList;

import retrofit2.Call;

public interface MovieRepo {

    Call<MovieList> getMovieWithId(
      String sorting
    );

    Call<Description> getDescriptionWithId(
      String id
    );

}

File: app/src/main/java/com/example/movieapi/data/MovieRepo.kt
package com.example.movieapi.data

import com.example.movieapi.api.Description
import com.example.movieapi.api.MovieList
import io.reactivex.Single
import retrofit2.Call

interface MovieRepo {
    fun getMovieWithId(
            sorting: String?
    ): Call<MovieList?>

    fun getDescriptionWithId(
            id: String?
    ): Single<Description>
}